Breeding-Insight/bi-api
# [BI-1465] 6.1 Observation Dataset Export File

# Description
**Story:** [BI-1465](https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?modal=detail&selectedIssue=BI-1465)

- `ExperimentController#datasetExport` was created to add the GET endpoint `/${micronaut.bi.api.version}/programs/{programId}/experiments/{experimentId}/export?{params}`
- `BrAPITrialService#exportObservations` and various helper functions in the same service were created to map existing observation data (if any) for the requested experiment and environments to table rows matching the structure of the experiment import template.
- `ExperimentControllerIntegrationTest` class was created to test the various combinations of query param values to the export endpoint.



# Dependencies
bi-web `feature/BI-1465`

# Testing

1. go to the Experiments and observations table
2. click Download for any experiment
3. select different file formats and the timestamp checkbox
4. verify fidelity of downloaded file to request inputs and import data for that experiment


# Checklist:
